main.tf

```diff
@@ -142,7 +142,7 @@ data "template_file" "gitlab_runner" {
     runners_token                           = var.runners_token
     secure_parameter_store_runner_token_key = local.secure_parameter_store_runner_token_key
     secure_parameter_store_region           = var.aws_region
-    gitlab_runner_registration_token        = var.gitlab_runner_registration_config["registration_token"]
```

Comment:
Is this really working? In this way the registration token is changed. It also seems the bug is solved in the gitlab runner: 
https://gitlab.com/gitlab-org/gitlab-runner/merge_requests/1144#9c5c461492d69bfd6a2ce40727e0b81b49a19e5b

Reply:
Ah, sorry. I just tested if the substitution is working. If Gitlab really fixed this. This PR is obsolete and can be closed. Thanks!

Reply:
I am going to test is today, just got a token with an underscore.

Reply:
I have just created a runner with a token containing an `_` the runner is created without an error and working as normal. Seems the issue is resolved by GitLab.